lib/datadog/tracing.rb

```diff
@@ -0,0 +1,198 @@
+module Datadog
+  # Datadog APM tracing public API.
+  #
+  # The Datadog teams ensures that public methods in this module
+  # only receive backwards compatible changes, and breaking changes
+  # will only occur in new major versions releases.
+  # @public_api
+  module Tracing
+    class << self
+      # (see Datadog::Tracer#trace)
+      # @public_api
+      def trace(name, continue_from: nil, **span_options, &block)
+        tracer.trace(name, continue_from: continue_from, **span_options, &block)
+      end
+
+      # (see Datadog::Tracer#continue_trace!)
+      # @public_api
+      def continue_trace!(digest, &block)
+        tracer.continue_trace!(digest, &block)
+      end
+
+      # The currently active {Datadog::Tracer} instance.
+      #
+      # The instance returned can change throughout the lifetime of the application.
+      # This means it is not advisable to cache it.
+      #
+      # The trace can be configured through {.configure},
+      # through {Datadog::Configuration::Settings::DSL::Tracer} options.
+      #
+      # TODO: This next paragraph can be better written.
+      #
+      # Most of the functionality available through the {.tracer} instance is
+      # also available in public methods in the {Datadog::Tracing} module.
+      # It is preferable to use the public methods in the {Datadog::Tracing} when possible
+      # as {Datadog::Tracing} strongly defines the tracing public API, and thus
+      # we strive to no introduce breaking changes to {Datadog::Tracing} methods.
+      #
+      # @return [Datadog::Tracer] the active tracer
+      # @!attribute [r] tracer
+      # @public_api
+      def tracer
+        components.tracer
+      end
+
+      # The tracer's internal logger instance.
+      # All tracing log output is handled by this object.
+      #
+      # The logger can be configured through {.configure},
+      # through {Datadog::Configuration::Settings::DSL::Logger} options.
+      #
+      # @!attribute [r] logger
+      # @public_api
+      def logger
+        Datadog.logger
+      end
+
+      # Current tracer configuration.
+      #
+      # To modify the configuration, use {.configure}.
+      #
+      # @return [Datadog::Configuration::Settings]
+      # @!attribute [r] configuration
+      # @public_api
+      def configuration
+        Datadog.configuration
+      end
+
+      # Apply configuration changes to `ddtrace`. An example of a {.configure} call:
+      # ```
+      # Datadog.configure do |c|
```

Comment:
Should this be `Datadog::Tracing.configure`?

Reply:
I think I was waiting for the Big Renaming PR to fix all these, but I can make this change in this PR.